### PR TITLE
Add com.tic80.TIC_80

### PR DESCRIPTION
**Pull request by kevinsmia1939**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:https://github.com/nesbox/TIC-80/pull/1957**
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*
  Metainfo and typo fix was submitted upstream https://github.com/nesbox/TIC-80/pull/1957
[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by kevinsmia1939** [comment]
bot, build com.tic80.TIC_80

File: com.tic80.TIC_80.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+           retro game.
+
+           Games are packaged into a cartridge file, which can be easily distributed.
+           TIC-80 works on all popular platforms. This means your cartridge can be
+           played in any device.
+
+           To make a retro styled game, the whole process of creation and execution
+           takes place under some technical limitations: 240x136 pixel display,
+           16 color palette, 256 8x8 color sprites, 4 channel sound, etc.
+        </p>
```
Comment (hfiguiere) [inline com.tic80.TIC_80.metainfo.xml:25]:
```suggestion
        <p>
           TIC-80 is a free and open source fantasy computer for making, playing
           and sharing tiny games.
        </p>
        <p>
           With TIC-80 you get built-in tools for development: code, sprites, maps,
           sound editors and the command line, which is enough to create a mini
           retro game.
        </p>
        <p>
           Games are packaged into a cartridge file, which can be easily distributed.
           TIC-80 works on all popular platforms. This means your cartridge can be
           played in any device.
        </p>
        <p>
           To make a retro styled game, the whole process of creation and execution
           takes place under some technical limitations: 240x136 pixel display,
           16 color palette, 256 8x8 color sprites, 4 channel sound, etc.
        </p>
```

File: com.tic80.TIC_80.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        commit: b09c50c61f2e8f66959ee9539e5a05feaeaf8ae1
+        x-checker-data:
+          type: git
+          tag-pattern: ^(v[\d.]+)$
+
+  - name: install-metainfo
+    buildsystem: simple
+    build-commands:
+      - install -Dm0644 $FLATPAK_ID.metainfo.xml ${FLATPAK_DEST}/share/metainfo/$FLATPAK_ID.metainfo.xml
+    sources:
```
Comment (hfiguiere) [inline com.tic80.TIC_80.yaml:37]:
```suggestion
      - install -Dm0644 $FLATPAK_ID.metainfo.xml ${FLATPAK_DEST}/share/metainfo/$FLATPAK_ID.metainfo.xml
    sources:
      - type: git
        url: https://github.com/nesbox/TIC-80.git
        tag: v1.0.2164
        commit: b09c50c61f2e8f66959ee9539e5a05feaeaf8ae1
        x-checker-data:
          type: git
          tag-pattern: ^(v[\d.]+)$
```

File: com.tic80.TIC_80.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+rename-desktop-file: tic80.desktop
+finish-args:
+  - --share=ipc
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --socket=pulseaudio
+  - --device=dri
+modules:
+  - name: TIC-80
+    buildsystem: cmake
```
Comment (hfiguiere) [inline com.tic80.TIC_80.yaml:15]:
```suggestion
    buildsystem: cmake-ninja
```

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by kevinsmia1939** [comment]
Thanks, cleanup a bit

**Comment by kevinsmia1939** [comment]
bot, build com.tic80.TIC_80

File: com.tic80.TIC_80.yaml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --socket=pulseaudio
+  - --device=dri
+modules:
+  - name: TIC-80
+    buildsystem: cmake-ninja
+    builddir: true
+    config-opts:
+      - -DBUILD_PRO=On
```
Comment (barthalion) [inline com.tic80.TIC_80.yaml:18]: I don't think this is fair, at least until payments support on Flathub is available? The author sells the pro version for $10 on Itch and I believe it should be disabled in the Flathub build for now.

What's your opinion about it @nesbox?

Comment (nesbox) [reply com.tic80.TIC_80.yaml:18]: I think PRO should be disabled by default because we use PRO payments to support TIC-80.

Comment (kevinsmia1939) [reply com.tic80.TIC_80.yaml]: ok, I removed the pro version.

**Comment by kevinsmia1939** [comment]
bot, build com.tic80.TIC_80

**Pull request closed without merging**